### PR TITLE
NMS-14012: DCB end-to-end smoke test

### PR DESCRIPTION
**Pull request by fooker**
### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14012

### CODE REVIEW

File: smoke-test/pom.xml [hunk 1]
```diff
@@ -634,6 +639,18 @@
...
+    <dependency>
+      <groupId>org.opennms.features.device-config.persistence</groupId>
+      <artifactId>org.opennms.features.device-config.persistence.impl</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
+      <dependency>
+          <groupId>org.opennms.features.device-config</groupId>
+          <artifactId>org.opennms.features.device-config.rest</artifactId>
+          <version>30.0.0-SNAPSHOT</version>
```
Comment (cgorantla) [inline smoke-test/pom.xml:651]: should be project.version
```suggestion
          <version>${ project.version}</version>
```

File: smoke-test/src/test/java/org/opennms/smoketest/dcb/DcbEndToEndIT.java [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+
+        // Ensure backup was not stored
+//        await().atMost(2, MINUTES)
+//               .until(restClient::getBackups, is(nullValue()));
+        // TODO: remove with NMS-14176
+        await().atMost(2, MINUTES)
+               .until(restClient::getBackups, jsonArray(hasSize(2)));
+    }
+
+    @Test
```
Comment (cgorantla) [inline smoke-test/src/test/java/org/opennms/smoketest/dcb/DcbEndToEndIT.java:357]: should add @Ignore until implemented

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml [hunk 3]
```diff
@@ -353,14 +353,14 @@
...
          <parameter key="username" value="${requisition:dcb:username|admin}"/>
          <parameter key="password" value="${requisition:dcb:password|admin}"/>
-         <parameter key="ssh-port" value="${requisition:ssh-port|22}"/>
-         <parameter key="ssh-timeout" value="${requisition:ssh-timeout|60000}"/>
+         <parameter key="ssh-port" value="${requisition:dcb:ssh-port|22}"/>
+         <parameter key="ssh-timeout" value="${requisition:dcb:ssh-timeout|60000}"/>
          <!-- schedule is a cron expression -->
          <parameter key="schedule" value="${requisition:dcb:schedule|0 0 0 * * ?}"/>
-         <parameter key="script-file" value="${requisition:script-file|assets:operating-system|node:os|default}.dcb"/>
+         <parameter key="script-file" value="${requisition:dcb:script-file|assets:operating-system|node:os|default}.dcb"/>
```
Comment (cgorantla) [inline opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml:363]: need to update these fields in the doc

**Review by cgorantla — COMMENTED** [review]
LGTM, minor comments

File: smoke-test/pom.xml (hunk 1 above)
Comment (fooker) [reply smoke-test/pom.xml]: Done

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml (hunk 3 above)
Comment (fooker) [reply opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml:363]: Done

File: smoke-test/src/test/java/org/opennms/smoketest/dcb/DcbEndToEndIT.java (hunk 2 above)
Comment (fooker) [reply smoke-test/src/test/java/org/opennms/smoketest/dcb/DcbEndToEndIT.java:358]: Done

**Review by christianpape — APPROVED** [review]
LGTM.

**Review by cgorantla — APPROVED** [review]

**Pull request merged**